src/dev/flang/fuir/analysis/AbstractInterpreter.java

```diff
@@ -610,6 +655,12 @@ public RESULT process(int cl, boolean pre, Stack<VALUE> stack, int c, int i)
         say("process "+_fuir.clazzAsString(cl)+"."+c+"."+i+":\t"+_fuir.codeAtAsString(cl, c, i)+" stack is "+stack);
       }
     var s = _fuir.codeAt(c, i);
+
+    _processor._fuir = _fuir;
```

Comment:
Since this only needed for obtaining the SourcePosition I think there should be an option/setting for the Processor to choose if we really need this in the current processor. E.g. for the Interpreter2 I may want to turn this off.

Reply:
I do not think this is very expensive, but I would like to further reduce the overhead of this as follows

- `c`/`i` should be joined into a `site` id, see #2473 
- `cl`/`pre` should become a single `int`, not sure if that should be a clazz id or something new 
- maybe, the `site` id could also contain combined `cl`/`pre`. Currently, I think that clazzes that origininate in the same feature share their code blocks, but it might make sense to replicate the code blocks to get unique `site` id. 
- Finally, if  `Processor` is not static class `_fuir` could beremoved. 

So, eventually, we might end up with a single `int` to identify the site.